### ISSUE TITLE
Oscillating loss on training set

### ISSUE THREAD

**Issue by wendlerc** [BVLC/caffe#744]
Hello,

Currently I am trying to apply the imagenet model to a simple image segmentation task, based on superpixel classification. However when I tune imagenet on the superpixel dataset, it seems like it learns almost nothing even after 50 000 iterations the loss on the training set is around 50%. In this case the database had 154982 foreground and 217205 backgound superpixel images. (superpixel image ~~ superpixel on a black background)

**Has anybody an idea what could go wrong?**

![Oscillating loss](http://image-upload.de/file/C1oXCK/b1a0fc1470.png)
![Accuracy](http://image-upload.de/file/VxKFGN/9481e5ba90.png)

Solver parameters

```
test_iter: 1000
test_interval: 500
base_lr: 0.001
lr_policy: "step"
gamma: 0.1
stepsize: 20000
display: 20
max_iter: 100000
momentum: 0.9
weight_decay: 0.0005
snapshot: 10000
```

The only thing that I changed in the network definition was the last layer:

```
layers {
  name: "fc8_aero"
  type: INNER_PRODUCT
  bottom: "fc7"
  top: "fc8_aero"
  blobs_lr: 1
  blobs_lr: 2
  weight_decay: 1
  weight_decay: 0
  inner_product_param {
    **num_output: 2**
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 0
    }
  }
}
```

Best regards,
Chris
